### DISCUSSION TITLE
"ffmpeg_recoverable_errors" - subset of the error?

### DISCUSSION THREAD

**Discussion by gniknalu** [roflcoopter/viseron#207] [Q&A] [answered]
Quick Question: For "ffmpeg_recoverable_errors" - do you put just a subset of the error string, the beginning of the string or something else?

I tried:
```
    ffmpeg_recoverable_errors:
      - Could not get needed stream information
```
But not sure if it is the correct thing for this error:

```
Exception in thread setup.BackWest:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/src/viseron/__init__.py", line 144, in run
    FFMPEGNVR(camera_config, self._detector)
  File "/src/viseron/nvr.py", line 125, in __init__
    self.camera = FFMPEGCamera(config, detector)
  File "/src/viseron/camera/__init__.py", line 34, in __init__
    self.initialize_camera(detector)
  File "/src/viseron/camera/__init__.py", line 59, in initialize_camera
    self.stream = Stream(
  File "/src/viseron/camera/stream.py", line 78, in __init__
    raise StreamInformationError(self.width, self.height, self.fps)
viseron.exceptions.StreamInformationError: Could not get needed stream information. Missing atleast one of width, height or fps. Width: 2592 Height: 1944 FPS: 0
```
I have two cameras that are slow to respond initially (because of extra hops) but once they get going, they're fine. Trying to get past that initial error.

**Thank you**

**Comment by roflcoopter** [comment, marked as answer]
`ffmpeg_recoverable_errors` will not help in this regard unfortunately.

Some cameras doesnt report their FPS correctly, (as you can see from the log FFprobe found Width, Height but no FPS, so you could just add `fps: CAMERA_FPS_HERE` to your camera config.

If you specify `width`, `height`, `fps`, `codec` and `audio_codec` in the config, then the call to FFprobe to get stream information will be skipped entirely and startup will be much faster, but there is a bug that causes Viseron to crash if you set `audio_codec` which i am going to push a fix for in a few minutes.
